### PR TITLE
Do real checks against the possible file extensions of a pkgfile

### PR DESCRIPTION
**Pull request by eli-schwartz**
See various discussion around https://lists.archlinux.org/pipermail/arch-projects/2018-February/004746.html and https://lists.archlinux.org/pipermail/arch-projects/2018-February/004748.html

It may make sense to add some version of https://lists.archlinux.org/pipermail/arch-projects/2018-February/004794.html in after the second commit.

@LukeShu

### CODE REVIEW

**Comment by LukeShu** [comment]
As I said on the ML, I'd have split up the final commit into 2 separate commits, because it's doing 2 different things:
 1. Fixing breakage in PKGEXT being a glob at all
 2. Switching PKGEXT to be an extended glob

Anyway, that's not such a big deal. LGTM (for what it's worth).

**Comment by eli-schwartz** [comment]
So this seems to sort of work. @LukeShu, were there any other patches you wanted in right now?

File: config [hunk 1]
```diff
@@ -25,7 +25,8 @@ TMPDIR="/var/tmp"
 ARCHES=(x86_64)
 DBEXT=".db.tar.gz"
 FILESEXT=".files.tar.gz"
-PKGEXTS=".pkg.tar.?z"
+# bash glob listing allowed extensions. Note that db-functions turns on extglob.
+PKGEXTS=".pkg.tar.@(gz|bz2|xz|lzo|lrz|Z)"
 SRCEXT=".src.tar.gz"
```
Comment (eli-schwartz) [inline config]: I forgot that I wanted to change the order here as @LukeShu suggested.

**Comment by pierres** [comment]
Looks fine to me. Also special thanks for the great commit messages which made it easy to follow your changes!

**Pull request merged**